### PR TITLE
Installation guide review

### LINKED ISSUE #112: DOC review installation guide

### LINKED ISSUE #119: DOC conda details should move to reference
The document titled [_Conda virtual environment_](https://bcda-aps.github.io/bluesky_training/instrument/_conda_environment.html) seems out of place in the `instrument/` directory.  There is a companion document titled [_Conda Base Environment_](https://bcda-aps.github.io/bluesky_training/reference/_conda_base.html) in `reference`.

These documents are linked from others.  The edit will involve more than just one file.  Maybe combine these two?

### LINKED ISSUE #127: installation steps need some updates
Just installing new instrument for HEXM and identified some matters for update.

1. Under **Setup a bluesky instrument**:

`/APSshare/bin/new_bluesky_instrument.py` would not run.

  - Tried `python /APSshare/bin/new_bluesky_instrument.py` but user account had Python2.7 as default.
  - Tried `python3 /APSshare/bin/new_bluesky_instrument.py` but `requests` package was not installed.
  - Tried `/APSshare/miniconda/x86_64/bin/python /APSshare/bin/new_bluesky_instrument.py` but the workstation could not reach web sites outside the firewall.
  - Finished the installation from a workstation with outside access.

2. **Create the conda environment**

  - Also need to run from a workstation with outside access.

3. This is a good point to **Start version control**

- create the GitHub organization (if the group does not already have one)
- create the GitHub repo
- clone the repo locally
- copy the `.git/` folder from the clone into `~/bluesky/` (this makes  `~/bluesky/`  a repo)
- `git add` and `git commit` the existing content in the new `bluesky` directory (before _any_ configuration changes are made)
- Need to get a `.gitignore` file (`new_bluesky_instrument.py` script should do this)

5. **Activate the conda environment**

  - no `conda` command, need to get that first:  `source /APSshare/miniconda/x86_64/bin/activate`

6. **Test the new bluesky instrument**

- Must create the **IPython profile** before this step.
- Should **Setup your databroker catalog configuration** before this step
- Needed to edit the `blueskyStarter.sh` file.
  - Specific changes in the new repo: https://github.com/APS-1ID-MPE/hexm-bluesky/commit/f71495c9fa68e48b43e6b2431bc64763b2c270ca
  - must be executable: `chmod +x blueskyStarter.sh`

With these changes, the instrument package started.

### LINKED ISSUE #139: DOC _testing.md
Name is not very meaningful. 
Contain both:
- ophyd testing with S:SRcurrentAI => redundant with _test_bluesky_at_APS.md
- APS data management   => could be a separate file (not every beamline are setup for the DM system)
I want to re-purpose _testing.md to only address the APS data management, and merge the ophyd part with _test_bluesky_at_APS.md

### PR DESCRIPTION
**Pull request by rodolakis**
- close #112
- close #139
- close #119
- close #127

### CODE REVIEW

File: docs/source/instrument/_install_new_instrument.rst [hunk 1]
```diff
@@ -0,0 +1,301 @@
+Installation
```
Comment (prjemian) [inline docs/source/instrument/_install_new_instrument.rst:1]: change title to `Install New Instrument` so the sidebar entry matches

extend the underline to match new title length

Comment (prjemian) [reply docs/source/instrument/_install_new_instrument.rst:1]: The rest of this document looks great!

File: docs/source/instrument/instrument_template.md [hunk 2]
```diff
@@ -3,5 +3,5 @@
 TODO: gather all that documentation here
```
Comment (prjemian) [inline docs/source/instrument/instrument_template.md:3]: Is this file necessary now?  Could be removed (any links?) with consequence.

File: docs/source/reference/_create_conda_env.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+Standard Library.  Rather than add these directly into the system Python
+installation, it is recommended to create a Python virtual environment which has
+the suite of packages (and specific versions) required.
+
+Follow these steps to create a conda environment for data collection with
+bluesky.
+
+**IMPORTANT**:  You must use `bash` shell.
+
+TODO: add how to get the conda command (for APSshare and without)
```
Comment (prjemian) [inline docs/source/reference/_create_conda_env.md:13]: TODO item

Comment (prjemian) [reply docs/source/reference/_create_conda_env.md:13]: Here are some places to look for this same content:

```bash
(bluesky_2023_2) .../bluesky_training$ git grep source | grep activate
bluesky/blueskyStarter.sh:                    source "$(dirname ${CMD})/activate" base
bluesky/run_qstarter_py.sh:source "${CONDA_BASE_DIR}/activate" "${ENV_NAME}"
docs/source/reference/_conda_base.md:source "${INSTALL_DIR}/bin/activate"
docs/source/reference/_conda_environment.md:   $ <b>source /PATH/TO/CONDA/bin/activate</b>
docs/source/tutor/_basic_a.ipynb:    "$ <b>source /APSshare/anaconda3/Bluesky/bin/activate</b>\n",
docs/source/tutor/_lesson1.ipynb:    "jemian@otz ~ $ source /APSshare/anaconda3/Bluesky/bin/activate \n",
install.md:    source $HOME/Apps/miniconda/bin/activate
```

File: docs/source/reference/_FAQ.rst [hunk 4]
```diff
@@ -18,11 +18,20 @@ This is a list of commonly asked questions concerning Bluesky.
...
 
-    Visit `Getting started <https://bcda-aps.github.io/bluesky_training/tutor/_start_bluesky_session.html>`_
+    Visit `Getting started <https://bcda-aps.github.io/bluesky_training/instrument/_getting_started.html>`_
+
+.. _faq-alias-start-bluesky:
+
+3. How to create an alias to start a bluesky session? 
+
+    There are several way to create an alias to start a bluesky session. One is described 
+    `here <https://bcda-aps.github.io/bluesky_training/instrument/_install_new_instrument.html#Create-a-bluesky-ipython-profile>`_; other methods TODO.
```
Comment (prjemian) [inline docs/source/reference/_FAQ.rst:28]: TODO item

Comment (prjemian) [reply docs/source/reference/_FAQ.rst:28]: Why not remove this text for now?
`; other methods TODO`

File: docs/source/reference/_FAQ.rst [hunk 5]
```diff
@@ -35,8 +44,21 @@ This is a list of commonly asked questions concerning Bluesky.
...
+    To determine if you're using bash, you can open up a terminal on your computer and type **echo $SHELL**.
+    If the output is **/bin/bash** or something similar, then you're using the bash shell.
+    
+    If bash is not your default shell, type **bash** in a terminal and press Enter to start a new instance of the bash shell. You should see a new prompt indicating that you're now using the bash shell. You can now type in bash commands. 
+    Note that any changes you make to your environment variables or other system settings within this bash session will only apply to this session and will not persist after you close the session. To change your default shell, contact your IT support. 
+
 .. _faq-linux-tilde:
 
-4. What does the (~) mean in a path?
+6. What does the (**~**) mean in a path?
```
Comment (prjemian) [inline docs/source/reference/_FAQ.rst:61]: Suggest `<code><b>~</b></code>` here and other similar places in this file.

Comment (prjemian) [reply docs/source/reference/_FAQ.rst:61]: Looks like this: <code><b>~</b></code>

Comment (prjemian) [reply docs/source/reference/_FAQ.rst:61]: and <code><b>bash</b></code>

**Review by prjemian — COMMENTED** [review]
Minor adjustments as noted and one TODO item to resolve.

File: docs/source/instrument/instrument_template.md (hunk 2 above)
Comment (rodolakis) [reply docs/source/instrument/instrument_template.md:3]: git grep did not find any reference to it, removed.

File: docs/source/reference/_FAQ.rst [hunk 6]
```diff
@@ -53,12 +53,12 @@ This is a list of commonly asked questions concerning Bluesky.
     To determine if you're using bash, you can open up a terminal on your computer and type **echo $SHELL**.
```
Comment (prjemian) [inline docs/source/reference/_FAQ.rst:53]: same change here

File: docs/source/reference/_FAQ.rst [hunk 7]
```diff
@@ -53,12 +53,12 @@ This is a list of commonly asked questions concerning Bluesky.
...
     If the output is **/bin/bash** or something similar, then you're using the bash shell.
```
Comment (prjemian) [inline docs/source/reference/_FAQ.rst:54]: same change here

File: docs/source/reference/_create_conda_env.md [hunk 8]
```diff
@@ -47,12 +88,8 @@ export BLUESKY_CONDA_ENV=bluesky_2023_2
 alias become_bluesky='conda activate ${BLUESKY_CONDA_ENV}'
 ```
 
-More documentation about [conda](../reference/_conda_base.md) is available
-[elsewhere](../reference/_conda_environment.md) in this repository.
-
-TODO: how to create an alias to activate the bluesky environment (with access to APSshare vs not).
 
-## bash environment variables
+TODO: how to create an alias to activate the bluesky environment (with access to APSshare vs not) and add to FAQ.
```
Comment (prjemian) [inline docs/source/reference/_create_conda_env.md:92]: Here's what I use (assumes `conda` is available and `BLUESKY_CONDA_ENV` environment variable is defined):

```bash
alias become_bluesky='conda activate ${BLUESKY_CONDA_ENV}'
```

Comment (prjemian) [reply docs/source/reference/_create_conda_env.md:92]: Context: it's in my `~/.bash_aliases` file:

```bash
(base) jemian@otz ~/.../bluesky/environments $ grep becom ~/.bash_aliases 
alias become_bluesky='conda activate ${BLUESKY_CONDA_ENV}'
```

**Comment by rodolakis** [comment]
Fixed _create_conda_env. I was planning to do that later (in a separated conda dedicated branch), but it think is is in a decent shape as is for now.

File: docs/source/reference/_create_conda_env.md (hunk 8 above)
Comment (rodolakis) [reply docs/source/reference/_create_conda_env.md:92]: I know, this needs more context  and work (how to deal with bashrc, source the bashrc_aliases etc). I can do something similar to what I wrote for the start_bluesky script tomorrow but I have to be gone in 1min.

Comment (rodolakis) [reply docs/source/reference/_create_conda_env.md:92]: I found the subtitle confusing though. It is only one piece of a much bigger thing.

Comment (rodolakis) [reply docs/source/reference/_create_conda_env.md:92]: give me 5 min

Comment (rodolakis) [reply docs/source/reference/_create_conda_env.md]: Done. I remove the stuff about the broken script for now.

**Comment by prjemian** [comment]
I'll fix those two files from my end.

**Comment by prjemian** [comment]
Just came across a lot of references to `http://nsls-ii.github.io` which are really, really old.  In most (?all?) cases, those should be changed to the appropriate `https://blueskyproject.io` site.  Needs some though.  

I can take this on.

**Comment by prjemian** [comment]
Also will look for any changes from `http://` to <code>http<b>s</b>://</code>

**Review by prjemian — APPROVED** [review]
LGTM, we reviewed together
